front/src/applications/stdcm/components/LayersModal.tsx

```diff
@@ -38,14 +41,17 @@ const LAYERS = [
   { layer: 'speed_limits', icon: <MdSpeed style={{ width: '20px' }} className="mx-2" /> },
 ];
 
-type StdcmLayersModalProps = {
-  initialLayers: LayersSettings;
-  onChange: (args: LayersSettings) => void;
+type LayersModalProps = {
+  onChange: (settings: Partial<MapSettings>) => void;
+  compactModal: boolean;
 };
 
-const StdcmLayersModal = ({ initialLayers, onChange }: StdcmLayersModalProps) => {
+const LayersModal = ({ onChange, compactModal }: LayersModalProps) => {
   const { t } = useTranslation();
-  const [selectedLayers, setSelectedLayers] = useState<LayersSettings>(initialLayers);
+  const { layersSettings, mapStyle } = useMapSettings();
+  const [selectedLayers, setSelectedLayers] = useState<LayersSettings>(layersSettings);
+  const mapSettings = useMapSettings();
```

Comment:
You have 2 hooks useMapSettings here :) Can you do this instead ?
```suggestion
  const mapSettings = useMapSettings();
  const { layersSettings, mapStyle } = mapSettings;
  const [selectedLayers, setSelectedLayers] = useState<LayersSettings>(layersSettings);
```